src/manifolds/Euclidean.jl

```diff
@@ -132,6 +132,39 @@ embed(::Euclidean, p)
 
 embed!(::Euclidean, q, p) = copyto!(q, p)
 
+@doc raw"""
+    embed(::EmbeddedManifold{𝔽,Euclidean{𝔽},Euclidean{𝔽},ET}, p)
+
+A [`Euclidean`](@ref) `M` manifold can be embedded into a [`Euclidean`](@ref) `N`
+if the length of the array dimension and representation size is elementwise less or equal.
+The remainder is filles with zeros for sure.
+"""
+function embed(M::EmbeddedManifold{𝔽,Euclidean{n,𝔽},Euclidean{m,𝔽},ET}, p) where {n,m,𝔽,ET}
+    q = allocate(p, representation_size(M.embedding))
+    embed!(M, q, p)
+    return q
+end
+
+function embed!(
+    M::EmbeddedManifold{𝔽,Euclidean{nL,𝔽},Euclidean{mL,𝔽},ET},
+    q,
+    p,
+) where {nL,mL,𝔽,ET}
+    n = size(p)
+    ln = length(n)
+    m = size(q)
+    lm = length(m)
+    (length(n) > length(m)) &&
+        throw(DomainError("Invalid embedding, since Euclidean dimension ($(n)) is longer than embedding dimension $(m)."))
+    any(n .> m[1:ln]) &&
+        throw(DomainError("Invalid embedding, since Euclidean dimension ($(n)) has entry larger than embedding dimensions ($(m))."))
+    # put p into q
+    fill!(q, 0.0)
+    # fill „top left edge“ of q with p.
+    q[[1:ind_n for ind_n in n]..., ones(Int, lm - ln)...] .= p
```

Comment:
```suggestion
    q[n_ind -> map(Base.OneTo(ind_n), n)..., ntuple(_ -> 1, lm - ln)...] .= p
```
I haven't tested it but it should be fine (and type-stable).

Reply:
Thanks, type (in)stability is something I do not always see.

Reply:
Hm, I can't even compile this? It says `syntax: "..." expression outside`, so what are we missing here?

Reply:
Ah, you shuffled it a little bit, it's `q[map(n_ind -> Base.OneTo(ind_n), n)..., ntuple(_ -> 1, lm - ln)...]`.

Reply:
Right, yes, thanks for fixing it. One thing to remember about splatting (the ... thing) is to only splat tuples when you care about performance.

Reply:
Now that works, but with different fields is still a little tricky, just leaving out the inner Fs did not do the trick.

Reply:
... and this time I've mixed `n_ind` with `ind_n`.

Reply:
;) Long day for me already, should've seen that. But I also got the different fields to work.

Reply:
Nice :slightly_smiling_face: